src/lazy/text/raw/v1_1/reader.rs

```diff
@@ -203,50 +214,64 @@ pub trait MacroIdLike<'a>: Sized + Copy {
         self.as_macro_id_ref()
     }
 
-    fn resolve(self, macro_table: &'a MacroTable) -> IonResult<ResolvedId<'a>> {
+    fn resolve<'b: 'a>(&self, macro_table: &'b MacroTable) -> IonResult<MacroRef<'b>> {
         use MacroIdRef::*;
         let id = self.as_macro_id_ref();
-        match id {
+
+        let qualified_address = match id {
             LocalName(name) => {
-                if let Some(address) = macro_table.address_for_name(name) {
-                    return Ok(ResolvedId::new(Some(name), address));
-                };
-            }
-            LocalAddress(address) => {
-                if let Some(name) = macro_table
-                    .macro_at_address(address)
-                    .and_then(|macro_ref| macro_ref.name())
-                {
-                    return Ok(ResolvedId::new(Some(name), address));
-                }
-            }
-            SystemAddress(_system_address) => {
-                todo!("resolving qualified macro IDs")
```

Comment:
🪧 Because macro handles now store a `QualifiedAddress`, it's possible to handle this case during resolution.